spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/support/JpaRepositoryFactoryBean.java

```diff
@@ -45,10 +51,12 @@
 public class JpaRepositoryFactoryBean<T extends Repository<S, ID>, S, ID>
 		extends TransactionalRepositoryFactoryBeanSupport<T, S, ID> {
 
+	private @Nullable BeanFactory beanFactory;
 	private @Nullable EntityManager entityManager;
 	private EntityPathResolver entityPathResolver;
 	private EscapeCharacter escapeCharacter = EscapeCharacter.DEFAULT;
 	private JpaQueryMethodFactory queryMethodFactory;
+	private @Nullable Function<BeanFactory, QueryEnhancerSelector> queryEnhancerSelector;
```

Comment:
The name is misleading. `queryEnhancerSelectorSource` of course we could go with `queryEnhancerSelectorSelector` ;-)